src/Components/Aspire.RabbitMQ.Client/RabbitMQEventSourceLogForwarder.cs

```diff
@@ -153,12 +153,26 @@ IEnumerator IEnumerable.GetEnumerator()
 
                     return index switch
                     {
-                        1 => new("ex.Type", exData["Type"]),
-                        2 => new("ex.Message", exData["Message"]),
-                        3 => new("ex.StackTrace", exData["StackTrace"]),
-                        4 => new("ex.InnerException", exData["InnerException"]),
+                        1 => new("exception.type", exData["Type"]),
+                        2 => new("exception.message", exData["Message"] is string message ? GetRabbitMQExceptionMessage(message) : null),
+                        3 => new("exception.stacktrace", exData["StackTrace"] is string { Length: 0 } && exData["Message"] is string message ? GetRabbitMQExceptionStackTrace(message) : exData["StackTrace"]),
+                        4 => new("exception.innerexception", exData["InnerException"]),
                         _ => throw new UnreachableException()
                     };
+
+                    string GetRabbitMQExceptionMessage(string rawMessage)
+                    {
+                        // rabbitmq message contains both message + stack trace. only take the first line
+                        var firstNewLineIndex = rawMessage.IndexOf(Environment.NewLine, StringComparison.Ordinal);
+                        return firstNewLineIndex is -1 ? rawMessage : rawMessage[..firstNewLineIndex];
+                    }
+
+                    string? GetRabbitMQExceptionStackTrace(string rawMessage)
+                    {
+                        // rabbitmq message contains both message + stack trace. only take subsequent lines
+                        var firstNewLineIndex = rawMessage.IndexOf(Environment.NewLine, StringComparison.Ordinal);
+                        return firstNewLineIndex is -1 || firstNewLineIndex == rawMessage.Length - Environment.NewLine.Length ? null : rawMessage[(firstNewLineIndex + Environment.NewLine.Length)..];
+                    }
```

Comment:
```suggestion
```

Let's not do this here. We should fix it in the underlying RabbitMQ.Client library. I've logged https://github.com/rabbitmq/rabbitmq-dotnet-client/issues/1493 to get it fixed in the right place.

With the update to 6.8.1, the only issue is that the full exception ToString shows up in the root "Message" entry.

![image](https://github.com/dotnet/aspire/assets/8291187/8ef5e92f-cd0b-4217-a189-c1d599ad204f)


The stack trace now shows up correctly in the right spot with the new update.

Reply:
Gotcha, thanks. reverted that change - now the only change is renaming the properties